client/settings/deposits/index.js

```diff
@@ -0,0 +1,230 @@
+	{ label: __( '16th', 'woocommerce-payments' ), value: 16 },
+	{ label: __( '17th', 'woocommerce-payments' ), value: 17 },
+	{ label: __( '18th', 'woocommerce-payments' ), value: 18 },
+	{ label: __( '19th', 'woocommerce-payments' ), value: 19 },
+	{ label: __( '20th', 'woocommerce-payments' ), value: 20 },
+	{ label: __( '21st', 'woocommerce-payments' ), value: 21 },
+	{ label: __( '22nd', 'woocommerce-payments' ), value: 22 },
+	{ label: __( '23rd', 'woocommerce-payments' ), value: 23 },
+	{ label: __( '24th', 'woocommerce-payments' ), value: 24 },
+	{ label: __( '25th', 'woocommerce-payments' ), value: 25 },
+	{ label: __( '26th', 'woocommerce-payments' ), value: 26 },
+	{ label: __( '27th', 'woocommerce-payments' ), value: 27 },
+	{ label: __( '28th', 'woocommerce-payments' ), value: 28 },
+	{
+		label: __( 'Last day of the month', 'woocommerce-payments' ),
+		value: 31,
+	},
+];
+
+const CustomizeDepositSchedule = () => {
+	const [
+		depositScheduleInterval,
+		setDepositScheduleInterval,
+	] = useDepositScheduleInterval();
+	const [
+		depositScheduleWeeklyAnchor,
+		setDepositScheduleWeeklyAnchor,
+	] = useDepositScheduleWeeklyAnchor();
+	const [
+		depositScheduleMonthlyAnchor,
+		setDepositScheduleMonthlyAnchor,
+	] = useDepositScheduleMonthlyAnchor();
+
+	const handleIntervalChange = ( newInterval ) => {
+		switch ( newInterval ) {
+			case 'weekly':
+				setDepositScheduleWeeklyAnchor(
+					depositScheduleWeeklyAnchor || 'monday'
+				);
+				break;
+
+			case 'monthly':
+				setDepositScheduleMonthlyAnchor(
+					depositScheduleMonthlyAnchor || '1'
+				);
+				break;
+		}
+
+		setDepositScheduleInterval( newInterval );
+	};
+
+	return (
+		<>
+			<div className="schedule-controls">
+				<SelectControl
+					label={ __( 'Frequency', 'woocommerce-payments' ) }
+					value={ depositScheduleInterval }
+					onChange={ handleIntervalChange }
+					options={ [
+						{
+							value: 'daily',
+							label: __( 'Daily', 'woocommerce-payments' ),
+						},
+						{
+							value: 'weekly',
+							label: __( 'Weekly', 'woocommerce-payments' ),
+						},
+						{
+							value: 'monthly',
+							label: __( 'Monthly', 'woocommerce-payments' ),
+						},
+					] }
+				/>
+				{ 'monthly' === depositScheduleInterval && (
+					<SelectControl
+						label={ __( 'Date', 'woocommerce-payments' ) }
+						value={ depositScheduleMonthlyAnchor }
+						onChange={ setDepositScheduleMonthlyAnchor }
+						options={ monthlyAnchors }
+					/>
+				) }
+				{ 'weekly' === depositScheduleInterval && (
+					<SelectControl
+						label={ __( 'Day', 'woocommerce-payments' ) }
+						value={ depositScheduleWeeklyAnchor }
+						onChange={ setDepositScheduleWeeklyAnchor }
+						options={ daysOfWeek }
+					/>
+				) }
+			</div>
+			<p className="help-text">
+				{ 'monthly' === depositScheduleInterval &&
+					__(
+						'Deposits scheduled on a weekend will be sent on the next business day.',
```

Comment:
Since this is a 'string' variable and not a function, it might be more clear to rename this to `depositStatus` rather than `getDepositStatus`.

Reply:
Fixed in f174838.